src/gmpints.c

```diff
@@ -568,6 +571,117 @@ Obj ObjInt_Int8( Int8 i )
 #endif
 }
 
+Obj ObjInt_UInt8( UInt8 i )
+{
+#ifdef SYS_IS_64_BIT
+  return ObjInt_UInt(i);
+#else
+  if (i == (UInt4)i) {
+    return ObjInt_UInt((UInt4)i);
+  }
+
+  /* we need two limbs to store this integer */
+  assert( sizeof(mp_limb_t) == 4 );
+  Obj gmp = NewBag( T_INTPOS, 2 * sizeof(mp_limb_t) );
+  mp_limb_t *ptr = ADDR_INT(gmp);
+  ptr[0] = (UInt4)i;
+  ptr[1] = ((UInt8)i) >> 32;
+  return gmp;
+#endif
+}
+
+/**************************************************************************
+**
+** Convert GAP Integers to various C types -- see header file
+*/
+Int Int_ObjInt(Obj i) {
+    UInt sign = 0;
+    if (IS_INTOBJ(i))
+        return INT_INTOBJ(i);
+    // must be a single limb
+    if (TNUM_BAG(i) == T_INTPOS)
+        sign = 0;
+    else if(TNUM_BAG(i) == T_INTNEG)
+        sign = 1;
+    else
+        ErrorMayQuit("Conversion error, expecting an integer, not a %s",(Int)TNAM_OBJ(i),0);
+    if (SIZE_BAG(i) != sizeof(mp_limb_t))
+        ErrorMayQuit("Conversion error, integer too large", 0L, 0L);
+    UInt val = ADDR_INT(i)[0];
+#ifdef SYS_IS_64_BIT
+    if ((!sign && (val > INT64_MAX)) ||
+        (sign && (val > (UInt)INT64_MIN))) // works because of twos complement arithmetic
+#else
+    if ((!sign && (val > INT32_MAX)) ||
+        (sign && (val > (UInt)INT32_MIN))) // works because of twos complement arithmetic
+#endif
```

Comment:
This is, "of course" an overflow check. But not everybody may know that, and if you don't, then this code looks like arcane magic, and the "works because ..." comment is not helpful either, because the reader may not even know what is supposed to "work" here.

In short: perhaps prefix this with a comment like `// check whether the result will fit into an Int`?